colexec: add telemetry for changing the vectorized cluster setting

asubiotto opened the pull request:
Release justification: low-risk update to new functionality

Release note: None (no user-visible changes)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46566)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46566#-:-M3Hdvh6580tWDr1lYqv:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto and @jordanlewis)

___
*[pkg/sql/exec_util.go, line 189 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46566#-M3HdsGn-pW977w9Ajcw:-M3HdsGn-pW977w9Ajcx:b-onvk8q) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6296837b7619f654c765acccd72e4061420115b/pkg/sql/exec_util.go#L189)):*
> ```Go
> )
> 
> const VectorizeClusterSettingName = "sql.defaults.vectorize"
> ```

nit: this probably needs a comment.
___
*[pkg/sql/logictest/testdata/logic_test/vectorize_telemetry, line 7 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46566#-M3Hd_K-2ESNTcZfIaZ7:-M3Hd_K-2ESNTcZfIaZ8:bod3hew) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6296837b7619f654c765acccd72e4061420115b/pkg/sql/logictest/testdata/logic_test/vectorize_telemetry#L7)):*
> ```
> 
> statement ok
> SET CLUSTER SETTING sql.defaults.vectorize='on';
> ```

nit: unnecessary semicolons.


<!-- Sent from Reviewable.io -->

Commit:
colexec: add telemetry for changing the vectorized cluster setting

Release justification: low-risk update to new functionality

Release note: None (no user-visible changes)

asubiotto commented:
bors r=yuzefovich
